other/lru_cache.py

```diff
@@ -127,62 +245,81 @@ def __contains__(self, key: int) -> bool:
 
         return key in self.cache
 
-    def get(self, key: int) -> int | None:
+    def get(self, key: T) -> U | None:
         """
-        Returns the value for the input key and updates the Double Linked List. Returns
-        None if key is not present in cache
+        Returns the value for the input key and updates the Double Linked List.
+        Returns None if key is not present in cache
```

Comment:
I would thrown a `KeyError` but we will go with it as is.
```suggestion
        Returns the value for the input key and updates the Double Linked List
        or returns None if key is not present in cache.
```

Reply:
I agree that is a better interface.  I just documented it while figuring out which types things /actually/ use.  :)

Reply:
Added a note that throwing KeyError would be a better and more python interface.